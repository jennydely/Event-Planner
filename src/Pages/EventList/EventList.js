import React, { useContext } from 'react'
import styled from 'styled-components/macro'
import resetIcon from '../../img/resetIcon.svg'
import { sortEvents } from './services/sortEvents'
import { filterEvents } from './services/filterEvents'
import { getFilteredEventList } from './services/getFilteredEventList'
import { getAvailableYears } from './services/getAvailableYears'
import EventItem from './components/EventItem'
import { UserContext } from "../../providers/UserProvider";
import useDeepCompareEffect from 'use-deep-compare-effect'

export default function EventList({
  eventArray,
  categoryFilter,
  eventFilter,
  chosenEventListFilter,
  eventSearch,
  multipleEventSearch,
  onTicketCheckboxClick,
  handleReloadButtonClick,
}) {
  const events = sortEvents(eventArray, eventFilter)
  const categoryfilteredEvents = filterEvents(events, categoryFilter)
  const filteredEventList = getFilteredEventList(
    {
      eventArray,
      eventFilter,
      chosenEventListFilter,
      eventSearch,
      multipleEventSearch,
      categoryfilteredEvents
    }
  )
  const availableYears = getAvailableYears(eventFilter, filteredEventList)
  const user = useContext(UserContext);
  useDeepCompareEffect(() => {
  }, [filteredEventList])


  return (
    <>
      {filteredEventList.length > 0 ? (
        availableYears.map((year) => (
          <EventContainer key={year}>
            <EventYearHeadline>Events {year}</EventYearHeadline>
            {filteredEventList.map(
              (event) =>
                event.eventStartDate.slice(0, 4) === year && (
                  user ?
                    <EventItem
                      event={event}
                      key={event.id}
                      id={event.id}
                      filteredEventList={filteredEventList}
                      onTicketCheckboxClick={onTicketCheckboxClick}
                    ></EventItem>
                    :
                    (<EventItem
                      event={event}
                      key={event.id}
                      id={event.id}
                    ></EventItem>)

                )
            )}
          </EventContainer>
        ))
      ) : (
          <>
            <p>No events</p>
            <ResetButton onClick={handleReloadButtonClick}>
              <img src={resetIcon} alt="reset" />
            </ResetButton>
          </>
        )}
    </>
  )
}

const EventContainer = styled.ul`
  margin: 0;
  padding: 0;
`
const EventYearHeadline = styled.h1`
  margin-bottom: 5px;
  padding: 0;
  text-align: left;
`
const ResetButton = styled.button`
  align-self: center;
`
