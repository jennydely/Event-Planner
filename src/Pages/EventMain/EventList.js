import React from 'react'
import styled from 'styled-components/macro'
import { sortEvents } from './services/sortEvents'
import { filterEvents } from './services/filterEvents'
import { getFilteredEvents } from './services/getFilteredEvents'
import { getAvailableYears } from './services/getAvailableYears'
import EventItem from './components/EventItem'

export default function EventList({
  eventArray,
  categoryFilter,
  eventFilter,
  eventSearch,
  multipleEventSearch,
  onHideButtonClick,
  onEditButtonClick,
  onDeleteButtonClick,
  onTicketCheckboxClick,
}) {
  const events = sortEvents(eventArray, eventFilter)
  const categoryfilteredEvents = filterEvents(events, categoryFilter)
  const filteredEvents = getFilteredEvents(
    eventArray,
    eventFilter,
    eventSearch,
    multipleEventSearch,
    categoryfilteredEvents
  )
  const availableYears = getAvailableYears(eventFilter, filteredEvents)

  return (
    <>
      {availableYears.map((year) => (
        <EventContainer key={year}>
          <EventYearHeadline>Events {year}</EventYearHeadline>
          {filteredEvents.map(
            (event) =>
              event.eventStartDate.slice(0, 4) === year && (
                <EventItem
                  event={event}
                  key={event.id}
                  id={event.id}
                  onHideButtonClick={onHideButtonClick}
                  onDeleteButtonClick={onDeleteButtonClick}
                  onEditButtonClick={onEditButtonClick}
                  onTicketCheckboxClick={onTicketCheckboxClick}
                ></EventItem>
              )
          )}
        </EventContainer>
      ))}
    </>
  )
}

const EventContainer = styled.ul`
  margin: 0;
  padding: 0;
`
const EventYearHeadline = styled.h1`
  margin-bottom: 5px;
  padding: 0;
  text-align: left;
`
