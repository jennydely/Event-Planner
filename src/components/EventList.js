import React from 'react'
import EventItem from './EventItem'
import styled from 'styled-components/macro'
import { sortEvents } from '../services/sortEvents'
import { filterEvents } from '../services/filterEvents'
import { getYearsOfEvents } from '../services/getYearsOfEvents'

export default function EventList({ eventArray, categoryFilter, eventFilter }) {
  const events = sortEvents(eventArray, eventFilter)
  const filteredEvents = filterEvents(events, categoryFilter)
  const availableYears = getYearsOfEvents(filteredEvents)

  return (
    <>
      {availableYears.map((year) => (
        <EventContainer key={year}>
          <EventYearHeadline>Events {year}</EventYearHeadline>
          {filteredEvents.map(
            (event) =>
              event.eventStartDate.slice(0, 4) === year && (
                <EventItem event={event} key={event.id}></EventItem>
              )
          )}
        </EventContainer>
      ))}
    </>
  )
}

const EventContainer = styled.ul`
  display: grid;
  grid-template-columns: auto 50px;
  margin: 10px;
  padding: 0;
`
const EventYearHeadline = styled.h1`
  margin-bottom: 5px;
  padding: 0;
  text-align: left;
`
