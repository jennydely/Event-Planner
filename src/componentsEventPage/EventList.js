import React from 'react'
import EventItem from './EventItem'
import styled from 'styled-components/macro'

export default function EventList({eventArray, categoryFilter}) {
    eventArray.sort((event1, event2) => event1.eventStartDate > event2.eventStartDate)
    const filteredEvents = eventArray.filter(event => categoryFilter === 'all' || event.category === categoryFilter)
    const currentYearString = new Date().getFullYear().toString()
    const availableYears = filteredEvents.reduce((years, event) => {
        const eventYear = event.eventStartDate.slice(0, 4)
        if (!years.includes(eventYear) && eventYear >= currentYearString) years.push(eventYear)
        return years
    }, [currentYearString])

    return (
        <>
            {availableYears.map(year => (
                <EventContainer key={year}>
                    <EventYearHeadline>Event {year}</EventYearHeadline>
                    {filteredEvents.map(event => event.eventStartDate.slice(0, 4) === year &&
                        <EventItem event={event} key={event.id}></EventItem>
                    )}
                </EventContainer>
            ))}
        </>
    )
}

const EventContainer = styled.ul`
margin: 10px;
padding:0;
`
const EventYearHeadline= styled.h2`
margin: 0;
margin-bottom: 5px;
padding:0;
text-align: left;
`
