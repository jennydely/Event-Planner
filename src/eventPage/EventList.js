import React from 'react';
import Event from './Event'
//import {getEvents} from '../utils/services'

const jsonEvents = require('../mockDB/events.json')

export default function EventList() {

    return (
        <ul>
            {jsonEvents.map(event =>
                <Event event={event} key={event.id}></Event>
            )}
        </ul>
    )
}