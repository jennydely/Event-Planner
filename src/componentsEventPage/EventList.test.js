import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import EventList from './EventList'
import 'jest-styled-components'

describe('EventList', () => {
    const eventArray = [{
        "id": "123",
        "eventName": "this is the eventname",
        "eventLocation": "this is the eventlocation",
        "eventDate": "06/25/2020 - 06/28/2020",
        "category": "medieval"
    },
    {
        "id": "124",
        "eventName": "this is the eventname",
        "eventLocation": "this is the eventlocation",
        "eventDate": "06/25/2021 - 06/28/2021",
        "category": "medieval"
    }

]
    const categoryFilter = 'medieval'


    it('display the year and an example event', () => {
        const { getByText } = render(<EventList eventArray={eventArray} categoryFilter={categoryFilter} />)
        setTimeout(() => {
            expect(getByText(eventArray[0].eventName)).toBeInTheDocument()
            expect(getByText(eventArray[0].eventLocation)).toBeInTheDocument()
        }, 100)
    })
    it('renders correctly', () => {
        const tree = renderer.create(<EventList eventArray={eventArray} categoryFilter={categoryFilter} />)
        expect(tree).toMatchSnapshot()
    })
})