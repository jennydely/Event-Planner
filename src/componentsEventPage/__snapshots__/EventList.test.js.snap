// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventList renders correctly 1`] = `
.c0 {
  margin: 10px;
  padding: 0;
}

.c1 {
  margin: 0;
  margin-bottom: 5px;
  padding: 0;
  text-align: left;
}

<ul
  className="c0"
>
  <h2
    className="c1"
  >
    Event 
    2020
  </h2>
</ul>
`;
